play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -1133,6 +1185,26 @@ export const PS = new class extends PSModel {
 				this.update();
 				continue;
 			}
+
+			case 'j':
+			case 'J': {
+				if (!room) continue;
+				let showjoins = this.prefs.showjoins || {};
+				let serverShowjoins = showjoins[this.server.id] || {};
+				if (!serverShowjoins[room.id] && !serverShowjoins.global) continue;
+				room.receiveLine([`j`, args[1]]);
+				break;
+			}
+
+			case 'l':
+			case 'L': {
+				if (!room) continue;
+				let showjoins = this.prefs.showjoins || {};
+				let serverShowjoins = showjoins[this.server.id] || {};
+				if (!serverShowjoins[room.id] && !serverShowjoins.global) continue;
+				room.receiveLine([`l`, args[1]]);
+				break;
+			}
```

Comment:
These guys would be best implemented in `panel-chat`'s `receiveLine`. You can make them `return` instead of `break` to prevent `log` from getting them.

Reply:
The problem is that this interception is too early. If you do it here, `ChatUserList` won't be able to update the userlist. Doing it there also makes sure you can do it after you update the userlist.